### PR TITLE
[Playwright] general flaky fix

### PR DESCRIPTION
**Pull request by aweng98**
Checking all tests run on Test env. I found one test consistently fails, `tests/dsm/participant-list/pancan-diagnosis-type-picklist-search.spec.ts`.

Failure cause: Click to close Column dropdown is not working. Colum dropdown's position is changing while Playwright attempts to click.

See video on CircleCI [run](https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/25941/workflows/345cf223-54e3-4f7d-8c0f-5c6c2ae3ab33/jobs/82120/tests).

All tests pass on Test env on [CircleCI](https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/25960/workflows/ce210135-f100-4b24-a908-16e1a0a60fbb/jobs/82634).

### CODE REVIEW

File: playwright-e2e/dsm/component/filters/sections/customize-view.ts [hunk 1]
```diff
@@ -82,18 +84,18 @@ export class CustomizeView {
...
 
   private get columnsGroupXPath(): string {
-    return `//div[button[@data-toggle="dropdown"] and button[.//*[text()[normalize-space()="${this.activeColumnsGroup}"]]]]`;
+    return `xpath=//div[button[@data-toggle="dropdown" and normalize-space()="${this.activeColumnsGroup}"]]`
   }
 
-  private columnPathXPath(columnName: string): string {
-    return `/ul/li/mat-checkbox[label[.//*[text()[normalize-space()="${columnName}"]]]]`;
+  private columnCheckbox(columnName: string): Checkbox {
+    return new Checkbox(this.page, { root: this.columnsGroupXPath, label: columnName })
```
Comment (aweng98) [inline playwright-e2e/dsm/component/filters/sections/customize-view.ts:107]: Avoid duplicated code by using `Checkbox.ts`.

File: playwright-e2e/dsm/component/filters/sections/customize-view.ts [hunk 2]
```diff
@@ -82,18 +84,18 @@ export class CustomizeView {
...
+    return `xpath=//*[text()[normalize-space()="Customize View"]]/button`;
   }
 
   private async isPanelOpen(): Promise<boolean> {
     return await this.page.locator('.btn-group').count() >= 1;
   }
 
   private get columnsGroupXPath(): string {
-    return `//div[button[@data-toggle="dropdown"] and button[.//*[text()[normalize-space()="${this.activeColumnsGroup}"]]]]`;
+    return `xpath=//div[button[@data-toggle="dropdown" and normalize-space()="${this.activeColumnsGroup}"]]`
```
Comment (aweng98) [inline playwright-e2e/dsm/component/filters/sections/customize-view.ts:103]: Simpify xpath.

File: playwright-e2e/dsm/component/filters/sections/customize-view.ts [hunk 3]
```diff
@@ -39,28 +40,29 @@ export class CustomizeView {
...
     const columnsGroupButton = this.columnsGroupButton({nth});
-    !(await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.click());
+    !(await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.locator('xpath=//*[@class="caret"]').click());
   }
 
   private async closeColumnsGroup(opts: { nth?: number } = {}): Promise<void> {
     const { nth } = opts;
     const columnsGroupButton = this.columnsGroupButton({nth});
-    (await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.click());
+    (await this.isExpanded(columnsGroupButton)) && (await columnsGroupButton.locator('xpath=//*[@class="caret"]').click());
```
Comment (aweng98) [inline playwright-e2e/dsm/component/filters/sections/customize-view.ts]: Looks for button's child span element to click in order to expand or close columns group dropdown.

File: playwright-e2e/dss/pages/atcp/atcp-dashboard-page.ts [hunk 4]
```diff
@@ -26,7 +26,20 @@ export default class AtcpDashboardPage extends AtcpPageBase {
 
   async expandTable(): Promise<void> {
     await waitForNoSpinner(this.page);
-    return this.getTable().headerLocator().waitFor({ state: 'visible', timeout: 5000 })
-      .catch(async () => await this.page.locator('.participant-expandable button.participant-expandable__control').click());
+    const expand = this.page.locator('.participant-expandable button.participant-expandable__control');
```
Comment (aweng98) [inline playwright-e2e/dss/pages/atcp/atcp-dashboard-page.ts:29]: Don't know why, table expanded then quickly collapsed. Adding checks to ensure table remains open.

**Review by Rinchan20 — APPROVED** [review]

**Pull request merged**